### PR TITLE
Nano: add dataloader transformation for `quantize` method

### PR DESCRIPTION
**Pull request by TheaperDeng**
### 1. Why the change?
https://github.com/intel-analytics/BigDL/issues/6389

### 2. User API changes
Nothing is changed

### 3. Summary of the change 
for inc based quantization in `InferenceOptimizer.quantize`, a pre-step is done to detect and transform the dataloader from 
```python
class MultipleTensorDataset(Dataset):
    ...
    def __getitem__(self, idx):
        return self.x1[idx], self.x2[idx], self.y[idx]
```
to
```python
class MultipleTensorDataset(Dataset):
    ...
    def __getitem__(self, idx):
        return (self.x1[idx], self.x2[idx]), self.y[idx]
```
### 4. How to test?
- [x] Unit test

Need a new how-to page to guide users about the supported dataloader.

### CODE REVIEW

File: python/nano/src/bigdl/nano/utils/inference/pytorch/dataloader.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                else:
+                    # if multiple y are provided
+                    return tuple(res[:forward_args_len]), tuple(res[forward_args_len:])
+            return collate_fn
+
+        # deepcopy the dataloader so that the transformation will not pollute the original one
+        new_dataloader = deepcopy(dataloader)
+
+        # get forward method's parameter number
+        forward_args = get_forward_args(model)
```
Comment (rnwang04) [inline python/nano/src/bigdl/nano/utils/inference/pytorch/dataloader.py:40]: `forward_args = get_forward_args(model)`
This occurs two times. Maybe move it before 
`if _need_dataloader_type_transformation(model, dataloader):` and pass them into _need_xxx

File: python/nano/test/pytorch/tests/test_inference_pipeline_ipex.py [hunk 2]
```diff
@@ -259,3 +271,34 @@ def test_get_model_with_method_name(self):
             assert isinstance(model, PytorchIPEXJITModel)
         except:
             pass
+
+    def test_multiple_input_dataloader(self):
+        # will not run this test if torch < 1.10
+        if TORCH_VERSION_LESS_1_10:
```
Comment (rnwang04) [inline python/nano/test/pytorch/tests/test_inference_pipeline_ipex.py:277]: which skip for this version？

File: python/nano/test/pytorch/tests/test_inference_pipeline_ipex.py [hunk 3]
```diff
@@ -259,3 +271,34 @@ def test_get_model_with_method_name(self):
...
+        InferenceOptimizer.quantize(net,
+                                    accelerator="onnxruntime",
+                                    calib_dataloader=dataloader)
+
+        # int8-onnxruntime
+        InferenceOptimizer.trace(net,
+                                 accelerator="onnxruntime",
+                                 input_sample=dataloader)
+
+        # int8-openvino
```
Comment (rnwang04) [inline python/nano/test/pytorch/tests/test_inference_pipeline_ipex.py:300]: here just openvino.
Maybe add an int8-openvino test case.

File: python/nano/src/bigdl/nano/utils/inference/pytorch/dataloader.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        new_dataloader.collate_fn = tuple_collate_fn_wrapper(new_dataloader.collate_fn,
+                                                             forward_args_len)
+
+        return new_dataloader
+    return dataloader
+
+
+def _need_dataloader_type_transformation(model, dataloader):
+    # get forward method's parameter number
+    forward_args = get_forward_args(model)
```
Comment (rnwang04) [inline python/nano/src/bigdl/nano/utils/inference/pytorch/dataloader.py:53]: get_forward_args() has a potential risk for kwargs.
For example, the forward function of model looks like : 
`model(x, mask=None)`
and dataloader returns `(x, y)`.
This may cause bug, and you maybe add a ut for this case.

**Review by rnwang04 — COMMENTED** [review]
others LGTM.

File: python/nano/test/pytorch/tests/test_inference_pipeline_ipex.py (hunk 2 above)
Comment (TheaperDeng) [reply python/nano/test/pytorch/tests/test_inference_pipeline_ipex.py:277]: for 1.9 ipex, some strange issue might be raised here, since we will gradually close the support for 1.9 so I just skip it here :)

File: python/nano/test/pytorch/tests/test_inference_pipeline_ipex.py (hunk 3 above)
Comment (TheaperDeng) [reply python/nano/test/pytorch/tests/test_inference_pipeline_ipex.py:300]: openvino int8 for multiple input case could not run through due to a reason other than dataloader's format, will raise a separate issue and pr for this.

File: python/nano/src/bigdl/nano/utils/inference/pytorch/dataloader.py (hunk 1 above)
Comment (TheaperDeng) [reply python/nano/src/bigdl/nano/utils/inference/pytorch/dataloader.py]: change to run `get_forward_args` only 1 time

File: python/nano/src/bigdl/nano/utils/inference/pytorch/dataloader.py (hunk 4 above)
Comment (TheaperDeng) [reply python/nano/src/bigdl/nano/utils/inference/pytorch/dataloader.py:50]: It works well, just add a ut for this
